src/terminal/adapter/DispatchTypes.hpp

```diff
@@ -109,31 +111,30 @@ namespace Microsoft::Console::VirtualTerminal
             return _value >= 0;
         }
 
-        constexpr size_t value() const noexcept
+        constexpr VTInt value() const noexcept
         {
             return _value;
         }
 
-        constexpr size_t value_or(size_t defaultValue) const noexcept
+        constexpr VTInt value_or(VTInt defaultValue) const noexcept
         {
             return has_value() ? _value : defaultValue;
         }
 
-        template<typename T, std::enable_if_t<sizeof(T) == sizeof(size_t), int> = 0>
+        template<typename T, typename = std::enable_if_t<std::is_enum_v<T>>>
         constexpr operator T() const noexcept
         {
-            // For most selective parameters, omitted values will default to 0.
             return static_cast<T>(value_or(0));
         }
 
-        constexpr operator size_t() const noexcept
+        constexpr operator VTInt() const noexcept
         {
             // For numeric parameters, both 0 and omitted values will default to 1.
-            return has_value() && _value != 0 ? _value : 1;
+            return _value > 0 ? _value : 1;

```

Comment:
The reason I made this change is that I actually found it difficult/confusing to understand when what value is returned initially. I distinctly remember that I was debugging something and it took me a second to realize that `has_value() && _value != 0` basically means `_value > 0`. Afterwards I found it easier to understand what breakpoints I need to set and how VT integer parameters are processed.
Truth be told it only took me a few seconds to understand, but still I find this new code a lot more obvious personally.

Reply:
Of course I don't mind reverting this. 🙂

Reply:
I'm happy to let a third party decide, but I specifically avoided using `_value > 0` because I was trying to make the intent clearer (not successfully it seems). If we're going to keep the `> 0` change, though, I think it at least needs an updated comment, because it makes no sense now without also explaining that we're using negative numbers to represent omitted values.

Reply:
I agree that we need the comment at a minimum. I don't have a preference which way the code falls, but it needs to be clear negative numbers are intentional.